src/pykeen/models/unimodal/ermlp.py

```diff
@@ -100,17 +131,21 @@ def score_hrt(self, hrt_batch: torch.LongTensor) -> torch.FloatTensor:  # noqa:
         # Embedding Regularization
         self.regularize_if_necessary(h, r, t)
 
-        # Concatenate them
-        x_s = torch.cat([h, r, t], dim=-1)
+        # Apply interaction function
+        return score(h=h, r=r, t=t, mlp=self.mlp)
 
-        # Compute scores
-        return self.mlp(x_s)
-
-    def score_t(self, hr_batch: torch.LongTensor) -> torch.FloatTensor:  # noqa: D102
-        # Get embeddings
-        h = self.entity_embeddings(hr_batch[:, 0])
-        r = self.relation_embeddings(hr_batch[:, 1])
-        t = self.entity_embeddings.weight
+    def score_t(
+        self,
+        hr_batch: torch.LongTensor,
+        h: Optional[torch.FloatTensor] = None,
+        r: Optional[torch.FloatTensor] = None,
+        t: Optional[torch.FloatTensor] = None,
+    ) -> torch.FloatTensor:  # noqa: D102
+        if h is None and r is None and t is None:
```

Comment:
if one of these is not none, there will be a problem. Also, if all are not none but hr_batch is also passed, there should be an error raised. Alternatively, it might make sense to split this function into a `def _score_t_helper(h,r,t) -> torch.FloatTensor` that can be more directly called if the user has their own h,r,t, or if they have a hr_batch, they can call `score_t()`

Sorry if I'm missing the point - I'm not exactly sure what the advantage of exposing this functionality is